src/player.cpp

```diff
@@ -11426,6 +11426,11 @@ void player::add_known_trap( const tripoint &pos, const trap &t)
     }
 }
 
+bool player::is_sleeping() const
+{
+    return has_effect( effect_sleep );
```

Comment:
> Doesn't this exist somewhere else already?

Strange, but there is no such global. There is local const in [game.cpp](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/game.cpp#L1555) and there is `Creature::in_sleep_state` in [creature.cpp](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/creature.cpp#L1101-L1104). The latter refers to `lying_down` effect which is I believe is when you are trying to sleep and can hear sounds.